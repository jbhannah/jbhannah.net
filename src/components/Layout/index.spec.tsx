import { render, screen } from "@testing-library/react"
import * as React from "react"
import { Layout } from "."

describe("Layout", () => {
  beforeEach(() => render(<Layout data-testid="layout" />))

  it("renders correctly", () => {
    expect(screen.getByTestId("layout")).toMatchSnapshot()
  })
})
