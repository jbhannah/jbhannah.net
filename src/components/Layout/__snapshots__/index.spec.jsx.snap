// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  css="[object Object]"
>
  <header
    css="[object Object]"
  >
    <h1
      css="[object Object]"
    >
      <a
        aria-current="page"
        class=""
        css="[object Object]"
        href="/"
      >
        Test Site
      </a>
    </h1>
    <nav
      css="[object Object]"
    >
      <ul
        css="[object Object]"
      >
        <li
          css="[object Object]"
        >
          <a
            css="[object Object]"
            href="https://www.example.com"
            rel="noopener"
            target="_blank"
          >
            <img
              alt="Test Name on Test Service"
              css="[object Object]"
              src="test-file-stub"
            />
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    css="[object Object],[object Object]"
  />
  <footer
    css="[object Object],[object Object]"
  >
    <div>
      Copyright Â© 2019 
      <a
        aria-current="page"
        class=""
        css="[object Object]"
        href="/"
      >
        Test Site
      </a>
      . 
      <a
        css="[object Object]"
        href="https://creativecommons.org/licenses/by-sa/4.0/"
        rel="license noopener"
        target="_blank"
      >
        Some Rights Reserved
      </a>
      . 
      <a
        css="[object Object]"
        href="https://github.com/jbhannah/jbhannah.net"
        rel="noopener"
        target="_blank"
      >
        Source on GitHub
      </a>
      .
    </div>
  </footer>
</div>
`;

exports[`PureLayout renders correctly 1`] = `
<div
  css={
    Object {
      "@media only screen and (min-width: 59rem)": Object {
        "flexFlow": "row wrap",
      },
      "@media only screen and (min-width: 70rem)": Object {
        "maxWidth": "42rem",
      },
      "display": "flex",
      "flexDirection": "column",
      "margin": "0 auto",
      "maxWidth": "54rem",
      "minHeight": "100vh",
    }
  }
>
  <ForwardRef(render)
    styles={
      Object {
        ".gatsby-highlight": Object {
          "& pre[class*=language-]": Object {
            "backgroundColor": "transparent",
            "float": "left",
            "margin": 0,
            "minWidth": "100%",
            "overflow": "initial",
            "padding": "0 1em 0 0",
          },
          "backgroundColor": "#fdf6e3",
          "borderRadius": "0.3em",
          "marginBottom": "1.58em",
          "overflow": "auto",
          "padding": "1em",
        },
        ".gatsby-highlight-code-line": Object {
          "backgroundColor": "#feb",
          "borderLeft": "0.3em solid #f99",
          "display": "block",
          "marginLeft": "-1.2em",
          "marginRight": "-2.2em",
          "paddingLeft": "0.9em",
          "paddingRight": "1em",
        },
      }
    }
  />
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Test Site
    </title>
    <meta
      name="description"
    />
    <meta
      content="Test Site"
      property="og:title"
    />
    <meta
      property="og:description"
    />
    <meta
      content="website"
      property="og:type"
    />
    <meta
      content="test-file-stub"
      property="og:image"
    />
    <meta
      content="Test Site"
      property="og:image:alt"
    />
    <meta
      property="og:url"
    />
    <meta
      property="twitter:creator"
    />
  </HelmetWrapper>
  <Header
    socialLinks={
      Array [
        Object {
          "key": "Test Service",
          "link": "https://www.example.com",
          "name": "Test Name",
          "service": "Test Service",
        },
      ]
    }
    title="Test Site"
  />
  <main
    css={
      Array [
        Object {
          "margin": "0 auto",
          "maxWidth": "42rem",
          "padding": "0 1rem",
          "width": "100%",
        },
        Object {
          "@media only screen and (min-width: 59rem)": Object {
            "margin": 0,
          },
          "flexGrow": 1,
        },
      ]
    }
  />
  <Footer
    title="Test Site"
  />
</div>
`;
